baseplate/integration/__init__.py

```diff
@@ -5,6 +5,39 @@
 
 """
 from .wrapped_context import WrappedRequestContext
+
+
+class _LazyAttributesMixin(object):
+    """A mixin that allows instances to add a lazily evaluated attribute
+       through `add_lazy_property`.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(_LazyAttributesMixin, self).__init__(*args, **kwargs)
+        self._lazy_attributes = {}
+
+    # Called when an attribute is missing
+    def __getattr__(self, name):
+        try:
+            lazy_fn = self._lazy_attributes[name]
+        except KeyError:
+            return super(_LazyAttributesMixin, self).__getattr__(name)
```

Comment:
Out of curiosity, did this change affect the benchmarks much at all?

Reply:
Not any dramatic impact. I think it'd be hard to see something like this come through in my micro benchmark:
```
Time elapsed to create 100000 spans (20 clients, 0 touched) = 3.6920 seconds (36.92μs / per)
Time elapsed to create 100000 spans (20 clients, 1 touched) = 4.1572 seconds (41.57μs / per)
Time elapsed to create 100000 spans (20 clients, 2 touched) = 4.4702 seconds (44.70μs / per)
Time elapsed to create 100000 spans (20 clients, 3 touched) = 4.7819 seconds (47.82μs / per)
Time elapsed to create 100000 spans (20 clients, 4 touched) = 5.0547 seconds (50.55μs / per)
Time elapsed to create 100000 spans (20 clients, 5 touched) = 5.4793 seconds (54.79μs / per)
Time elapsed to create 100000 spans (20 clients, 6 touched) = 5.7219 seconds (57.22μs / per)
Time elapsed to create 100000 spans (20 clients, 7 touched) = 6.0422 seconds (60.42μs / per)
Time elapsed to create 100000 spans (20 clients, 8 touched) = 6.8233 seconds (68.23μs / per)
Time elapsed to create 100000 spans (20 clients, 9 touched) = 6.8550 seconds (68.55μs / per)
Time elapsed to create 100000 spans (20 clients, 10 touched) = 7.6404 seconds (76.40μs / per)
Time elapsed to create 100000 spans (20 clients, 11 touched) = 7.3861 seconds (73.86μs / per)
```

Reply:
Makes sense, thanks!